kgforge/specializations/stores/bluebrain_nexus.py

```diff
@@ -704,19 +641,16 @@ def search(
         includes = params.get("includes", None)
         excludes = params.get("excludes", None)
         search_endpoint = params.get(
-            "search_endpoint", self.service.sparql_endpoint["type"]
+            "search_endpoint",  Service.SPARQL_ENDPOINT_TYPE  # default search endpoint is sparql
         )
-        if search_endpoint not in [
-            self.service.sparql_endpoint["type"],
-            self.service.elastic_endpoint["type"],
-        ]:
-            raise ValueError(
-                f"The provided search_endpoint value '{search_endpoint}' is not supported. Supported "
-                f"search_endpoint values are: '{self.service.sparql_endpoint['type'], self.service.elastic_endpoint['type']}'"
-            )
-        if "filters" in params:
```

Comment:
Why remove this ?

Reply:
For the sake of usability, filters was *filters in the forge signature which made sense. However it was also *filters in the store signature, which doesn't make much sense in my opinion (users are not exposed to the store), so now it should be provided as a keyword parameter. So it's not possible for filters to ever make it to params, the provided value will be attributed to the explicit declaration of filters in the method signature

Reply:
see https://github.com/BlueBrain/nexus-forge/commit/4d2580e4b7c4514278fa7385109cee0d794a6b36#diff-9ee1d07daf4890e0d1087d9815baaea67a26d4fd6b31e546784b695165c8e68aR686

Reply:
I see. This change from *filters to filters introduced an API change and it is not clear to me why this change was necessary. I'll review the why this change in another PR before release. For now, you are right to align the implementation.

Reply:
Come to think of it, this check can be kept but it would be done in forge method, not the store method

Reply:
The way I see *filters is nothing more than a way to simplify the usage of user-exposed functions. However this list unpacking was spread all the way down to parts of the code that are not user-exposed (ex: Sparql query builder) and it was preventing the passing of arguments in "any order" (with keywords). In a way, it was forcing each method call to be aware of the position of the arguments to be provided, and that was very error-prone (it's easier to provide parameters with keywords, that are often documented, and checked by an IDE, than it is to trust a developer to provide them in the right order, without keywords)

Reply:
That could work. The API is what is in kgforge.core